### ISSUE TITLE
"affected" npm scripts error when trying to resolve dependencies

### ISSUE THREAD

**Issue by emilio-martinez** [nrwl/nx#183]
**Nx** 0.6
**Angular CLI** 1.5.0 (via `nrwl/fix-cli-build`. For other reasons, can't upgrade to 1.6.x yet)
**NPM** 5.5.1

---

When trying to run the "affected" scripts, e.g., `npm run build:affected -- --files="libs/gestures/index.ts"` or `npm run build:affected -- 4351dd56f3c4a71c4fe134f1caf35928f75080c5 e1ee7bb1b589058fa269d00e142b2b629d236819`, regardless of what script is used, it errors out on me. This is after upgrading from 0.5.3, which I don't know if it makes a difference.

Because of the `try...catch` in `run-affected.js:10`, the errors thrown for all three scripts aren't really reflective of the issue. Commenting the `try...catch` out, the stack tracks down to the recursive call in `Deps.prototype.transitiveDeps`.

Below I'm providing error stacks for both before and after commenting out the `try...catch` block, just in case somebody is searching for the uncommented version. FYI, whether I'm using SHAs or `files`, I still get the errors.

## Without `try...catch`

For all three scripts, i.e., `npm run apps:affected -- 4351dd56f3c4a71c4fe134f1caf35928f75080c5 e1ee7bb1b589058fa269d00e142b2b629d236819`, `npm run build:affected -- 4351dd56f3c4a71c4fe134f1caf35928f75080c5 e1ee7bb1b589058fa269d00e142b2b629d236819` and `npm run e2e:affected -- 4351dd56f3c4a71c4fe134f1caf35928f75080c5 e1ee7bb1b589058fa269d00e142b2b629d236819`, the error returend is

```
/project_root/node_modules/@nrwl/schematics/src/affected/affected.js:74
        this.deps[project].forEach(function (d) {
                           ^

RangeError: Maximum call stack size exceeded
    at Array.forEach (native)
    at Deps.transitiveDeps (/project_root/node_modules/@nrwl/schematics/src/affected/affected.js:74:28)
    at /project_root/node_modules/@nrwl/schematics/src/affected/affected.js:75:36
    at Array.forEach (native)
    at Deps.transitiveDeps (/project_root/node_modules/@nrwl/schematics/src/affected/affected.js:74:28)
    at /project_root/node_modules/@nrwl/schematics/src/affected/affected.js:75:36
    at Array.forEach (native)
    at Deps.transitiveDeps (/project_root/node_modules/@nrwl/schematics/src/affected/affected.js:74:28)
    at /project_root/node_modules/@nrwl/schematics/src/affected/affected.js:75:36
    at Array.forEach (native)
```

## With `try...catch`

The error for `npm run apps:affected -- 4351dd56f3c4a71c4fe134f1caf35928f75080c5 e1ee7bb1b589058fa269d00e142b2b629d236819` is
```
Pass the SHA range, as follows: npm run apps:affected SHA1 SHA2.
Or pass the list of affected files, as follows: npm run apps:affected --files="libs/mylib/index.ts,libs/mylib2/index.ts".
/project_root/node_modules/@nrwl/schematics/src/affected/run-affected.js:20
        console.log(apps.join(' '));
                        ^

TypeError: Cannot read property 'join' of undefined
    at Object.<anonymous> (/project_root/node_modules/@nrwl/schematics/src/affected/run-affected.js:20:25)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:504:3
```

The error for `npm run build:affected -- 4351dd56f3c4a71c4fe134f1caf35928f75080c5 e1ee7bb1b589058fa269d00e142b2b629d236819`
```
Pass the SHA range, as follows: npm run build:affected SHA1 SHA2.
Or pass the list of affected files, as follows: npm run build:affected --files="libs/mylib/index.ts,libs/mylib2/index.ts".
/project_root/node_modules/@nrwl/schematics/src/affected/run-affected.js:88
    if (apps.length > 0) {
            ^

TypeError: Cannot read property 'length' of undefined
    at build (/project_root/node_modules/@nrwl/schematics/src/affected/run-affected.js:88:13)
    at Object.<anonymous> (/project_root/node_modules/@nrwl/schematics/src/affected/run-affected.js:23:9)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
```

The error for `npm run e2e:affected -- 4351dd56f3c4a71c4fe134f1caf35928f75080c5 e1ee7bb1b589058fa269d00e142b2b629d236819`
```
Pass the SHA range, as follows: npm run e2e:affected SHA1 SHA2.
Or pass the list of affected files, as follows: npm run e2e:affected --files="libs/mylib/index.ts,libs/mylib2/index.ts".
/project_root/node_modules/@nrwl/schematics/src/affected/run-affected.js:99
    if (apps.length > 0) {
            ^

TypeError: Cannot read property 'length' of undefined
    at e2e (/project_root/node_modules/@nrwl/schematics/src/affected/run-affected.js:99:13)
    at Object.<anonymous> (/project_root/node_modules/@nrwl/schematics/src/affected/run-affected.js:26:9)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:389:7)
    at startup (bootstrap_node.js:149:9)
```

**Comment by emilio-martinez** [comment]
The repo this is occurring on is private, but if the code is useful to repro, happy to either duplicate repo and add collaborators, zip-up to share via a secured link, or whatever method is preferred.

**Comment by vsavkin** [comment]
I believe the issue has been fixed here: 3531323fb5210b995b1296a198c8e76ee8bf9a07 The fix hasn't been released yet.

**Comment by emilio-martinez** [comment]
Released on 0.6.10. Thank you @vsavkin!
